source/aggregation.txt

```diff
@@ -0,0 +1,195 @@
+finished product.
+
+The **aggregation pipeline** is the assembly line, **aggregation stages** are the
+assembly stations, and **operator expressions** are the
+specialized tools.
+
+Aggregation Versus Find Operations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use find operations to perform the following actions:
+
+- Select which documents to return
+- Select which fields to return
+- Sort the results
+
+You can use aggregation operations to perform the following actions:
+
+- Perform find operations
+- Rename fields
+- Calculate fields
+- Summarize data
+- Group values
+
+Limitations
+~~~~~~~~~~~
+
+Keep the following limitations in mind when using aggregation operations:
+
+- Returned documents must not violate the
+  :manual:`BSON document size limit </reference/limits/#mongodb-limit-BSON-Document-Size>`
+  of 16 megabytes.
+- Pipeline stages have a memory limit of 100 megabytes by default. You can exceed this
+  limit by setting the ``allow_disk_use`` field of a ``mongocxx::options::aggregate``
+  instance.
+
+.. important:: $graphLookup Exception
+
+   The :manual:`$graphLookup
+   </reference/operator/aggregation/graphLookup/>` stage has a strict
+   memory limit of 100 megabytes and ignores the ``allow_disk_use`` field.
+
+.. _cpp-aggregation-example:
+
+Aggregation Example
+-------------------
+
+.. note::
+  
+   The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+   database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+   free MongoDB Atlas cluster and load the sample datasets, see the :atlas:`Get Started with Atlas
+   </getting-started>` guide.
+
+To perform an aggregation, pass a ``mongocxx::pipeline`` instance containing the aggregation
+stages to the ``collection.aggregate()`` method.
+
+The following code example produces a count of the number of bakeries in each borough
+of New York. To do so, it uses an aggregation pipeline that contains the following stages:
+
+- :manual:`$match </reference/operator/aggregation/match/>` stage to filter for documents
+  in which the ``cuisine`` field contains the value ``"Bakery"``
+
+- :manual:`$group </reference/operator/aggregation/group/>` stage to group the matching
+  documents by the ``borough`` field, accumulating a count of documents for each distinct
+  value
+
+.. io-code-block::
+
+   .. input:: /includes/aggregation.cpp
+      :start-after: start-match-group
+      :end-before: end-match-group
+      :language: cpp
+      :dedent:
+
+   .. output:: 
+
+      { "_id" : "Brooklyn", "count" : 173 }
+      { "_id" : "Queens", "count" : 204 }
+      { "_id" : "Bronx", "count" : 71 }
+      { "_id" : "Staten Island", "count" : 20 }
+      { "_id" : "Missing", "count" : 2 }
+      { "_id" : "Manhattan", "count" : 221 }
+
+Explain an Aggregation
+~~~~~~~~~~~~~~~~~~~~~~
+
+To view information about how MongoDB executes your operation, you can
```

Comment:
S: in all the uses of MongoDB in this paragraph, is it possible to clarify what part of the MongoDB product is explaining an operation? ex: the query api